app/controllers/concerns/idv/verify_info_concern.rb

```diff
@@ -96,6 +103,9 @@ def idv_failure(result)
       elsif proofing_results_exception.present? && is_mva_exception
         idv_failure_log_warning
         redirect_to state_id_warning_url
+      elsif proofing_results_exception.present? && is_threatmetrix_exception
+        idv_failure_log_warning
+        redirect_to warning_url
```

Comment:
Having now spent a little more time with this, I think what we want to do here is tweak the priority of how we are redirecting the user.

First, if they are rate limited (by the SSN rate limiter or Resolution rate limiter), they should be redirected to the appropriate "rate limited" warning.

Next, if there is an AAMVA MVA exception, we should redirect to the special error screen we have for that (`state_id_warning_url`)

Next, if resolution failed but there is no exception present, we should display the "we could not match you to records" error.

Finally, if resolution failed due to an exception, we should display the "internal error" message.

This way failed resolution + a ThreatMetrix exception means the user sees the "could not match you to records" message rather than the "internal error" message, which is more actionable in this case.

Reply:
@matthinz  - I think I addressed this feedback (see this [diff range](https://github.com/18F/identity-idp/pull/11459/files/1d9643d2d7ef207b581f4512cf6defe6fa31a7a9..cc66d802ec7776a0f51f88c34c384e7677b56531)). Let me know what you think!

Reply:
I _think_ one case not covered here is if the user passed everything _except_ there was a Threatmetrix exception, then we want to show them the "Internal error, please try again" message.

In the terms used by the [ResultAdjudicator](https://github.com/18F/identity-idp/blob/cc66d802ec7776a0f51f88c34c384e7677b56531/app/services/proofing/resolution/result_adjudicator.rb#L31-L32), this would occur when:

* `device_profiling_success == false`
* `device_profiling_reason == :device_profiling_exception`
* `resolution_success == true`

We may want to consider adding a new thing to the adjudicated result, something like `user_could_fix_error`, which would indicate whether there is any chance the user could affect the resolution result by changing their PII. If no they _could_ fix the error, we show the "we could not match your information to records" message. Otherwise we show the internal error message.

Reply:
Is this not covered by `has_exception && is_threatmetrix_exception`? As far as I can tell in our analytics, when there is a ThreatMetrix API exception,  `device_profiling_reason` is always `:device_profiling_exception`. What am I missing?

Reply:
> We may want to consider adding a new thing to the adjudicated result, something like user_could_fix_error, which would indicate whether there is any chance the user could affect the resolution result by changing their PII. If no they could fix the error, we show the "we could not match your information to records" message. Otherwise we show the internal error message.

This feels like a separate ticket and should include a doc that maps failure state matrices to resolution action.

Reply:
Oof, I wrote the first half of that comment last week. Let me try and get my head back around this

Reply:
We resolved this offline and a different work item will be created to refactor the ResultAdjudicator to hold this logic. Will link here once made.